### DISCUSSION TITLE
22/02/18 RobotLibrary Development Meeting

### DISCUSSION THREAD

**Discussion by ssutjipto** [Woolfrey/software_robot_library#27] [Meetings]
### Chair: 
- Sheila @ssutjipto 

### Attendees:
- Jon @Woolfrey 
- Sheila @ssutjipto 

### Agenda:
| Item | Tag | Responsible | Comment |
|------|-----|---------------|---------------|
| Follow up on calcs with new `RigidyBody`, `Joint` classes. | #2 | @ssutjipto | Resolved with #24 |
| Add debugging support. | N/A | @ssutjipto | Still in development | 
| Add Null Space projection to control class(es) | N/A | @Woolfrey | Completed, singularity avoidance is used as default if arm is redundant |
| Create new repository for Gazebo / ROS simulation | N/A | @Woolfrey | Completed, but needs further development |
| Formalize meeting minutes somehow in gitub. | N/A | @Woolfrey | Template made and meeting discussion board created |
| Merge `main` < `devel` < `new` | #2 | @ssutjipto, @Woolfrey |Resolved with #24 |

### Additional Discussion Points:
| Topic | Raised By | Tag | Outcome |
|-------|------------|-----|------------|
| Debugging for RobotLibrary | @ssutjipto | N/A | A feature branch will be made where this will be developed. |
| Incorporating threading for control | @ssutjipto | N/A | For now we will rely on ROS nodes to synchronize data and control topics. |
| Development of a ROS wrapper for the RobotLibrary | @ssutjipto | N/A | Each robot has its own custom msg topics in ROS, so for now we will just use a custom node for each type of robot (UR3, Sawyer, Kuka, etc.) |

### Next Meeting:
| Item | Tag | Responsible | 
|------|-----|---------------|
| Follow up on nullspace projection, check J*N == 0 | N/A | @Woolfrey |
| Look into creating a new project board that spans the RobotLibrary and ROS package| N/A | @Woolfrey |
| Create new feature branch for debugging. | N/A | @ssutjipto |
| Add support and instructions for debugging | N/A | @ssutjipto |
| Joint limit avoidance in `Control` class | N/A | @Woolfrey |
| Demoting outstanding Issues to Notes in the project page | N/A | @Woolfrey |

### Conclusions:
- Create new project tab for managing ROS package development
- How to design structure of ROS interface when interacting with the RobotLibrary
- Rely on ROS publishers and subscribers instead of threading

**Comment by Woolfrey** [comment]
I put the "Additional Discussion Points" in to a table so it's a little bit more organised.

**Comment by Woolfrey** [comment]
Added an extra item for the next meeting on sorting out Issues, which I forgot to raise in the last meeting...
